File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1]*len(nums)
        pre = post = 1
        for i in range(0, len(nums)):
            res[i] *= pre
            pre *= nums[i]
        post = nums[len(nums)-1]
        for i in range(len(nums)-2, -1, -1):
            res[i] *= post 
            post *= nums[i]
        return res

File: Data Structures & Algorithms/special-array-i/submission-2.go
func isArraySpecial(nums []int) bool {
    i := 0
    j := 1

    for j < len(nums) {
        if nums[i] % 2 == nums[j] % 2 {
            return false
        }
        i += 1
        j += 1
    }

    return true
}